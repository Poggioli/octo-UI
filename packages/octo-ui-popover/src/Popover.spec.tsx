import { render } from "@testing-library/react";
import {
  Popover,
  PopoverAnchor,
  PopoverClose,
  PopoverCloseIcon,
  PopoverContent,
  PopoverTrigger,
} from "./Popover";

describe("<Popover />", () => {
  it(`GIVEN a <Popover />
      WHEN render is called
      THEN should render a <Popover />`, () => {
    const rendered = render(
      <Popover defaultOpen>
        <PopoverTrigger>
          <button>Trigger</button>
        </PopoverTrigger>
        <PopoverAnchor>
          <button>Anchor</button>
        </PopoverAnchor>
        <PopoverContent>
          <h3>Popover content</h3>
          <PopoverClose>
            <button>Close</button>
          </PopoverClose>
          <PopoverCloseIcon />
        </PopoverContent>
      </Popover>
    );
    expect(rendered.container).toMatchSnapshot();
  });

  it(`GIVEN a <Popover />
      WHEN toString is called
      THEN return a string`, () => {
    expect(Popover.toString()).toBe(".octo-UI-c-PJLV");
  });

  it(`GIVEN a <PopoverAnchor />
      WHEN toString is called
      THEN return a string`, () => {
    expect(PopoverAnchor.toString()).toBe(".octo-UI-c-PJLV");
  });

  it(`GIVEN a <PopoverClose />
      WHEN toString is called
      THEN return a string`, () => {
    expect(PopoverClose.toString()).toBe(".octo-UI-c-PJLV");
  });

  it(`GIVEN a <PopoverCloseIcon />
      WHEN toString is called
      THEN return a string`, () => {
    expect(PopoverCloseIcon.toString()).toBe(".octo-UI-c-PJLV");
  });

  it(`GIVEN a <PopoverContent />
      WHEN toString is called
      THEN return a string`, () => {
    expect(PopoverContent.toString()).toBe(".octo-UI-c-ewLfbo");
  });

  it(`GIVEN a <PopoverTrigger />
      WHEN toString is called
      THEN return a string`, () => {
    expect(PopoverTrigger.toString()).toBe(".octo-UI-c-PJLV");
  });
});
