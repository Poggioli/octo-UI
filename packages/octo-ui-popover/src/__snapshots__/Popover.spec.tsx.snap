// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Popover /> GIVEN a <Popover />
      WHEN render is called
      THEN should render a <Popover /> 1`] = `
<div>
  <button
    aria-controls="radix-:r0:"
    aria-expanded="true"
    aria-haspopup="dialog"
    class="octo-UI-c-PJLV"
    data-state="open"
    type="button"
  >
    Trigger
  </button>
  <button
    class="octo-UI-c-PJLV"
  >
    Anchor
  </button>
</div>
`;
